### PR TITLE
Add basic collapse/expand for pivot table row groups

### LINKED ISSUE #14171: Pivot tables: collapsable/expanding rows
The scope and spec for pivot tables includes a feature to allow collapsing a row that contains one or more subgroups, collapsing all the child groups and rows down and only showing the subtotal row for the collapsed group.

image

collapsable-rows

**Links**
- [Section in the main pivot tables product doc about this feature](https://docs.google.com/document/d/1ScNpNjjYlsi6syyHFKctTlZ149g5ulTIHFnAmmq6h58/edit#bookmark=id.ntxc95io2fmt)
- eng doc: …
- feature branch: …
- Issue links: …

**Implementation Plan**

**ETA**: …

- [ ] step 1: …

**Needs to happen before merge**

- [ ] …

### PR DESCRIPTION
**Pull request by paulrosenzweig**
This PR adds a new visualization setting to collapse row groups.

Closes https://github.com/metabase/metabase/issues/14171

collapse_expand

Here's the product doc image for context on style changes. I wasn't sure whether we want the icon background since the icon also appears over the subtotal background.
image

Also, It really seems like the open/close should be animated, but I didn't see a non-horrible way to accomplish this in the context of the virtualization library we're using.

If you checkout this branch, [this link](http://localhost:3000/question#eyJkYXRhc2V0X3F1ZXJ5Ijp7InR5cGUiOiJxdWVyeSIsInF1ZXJ5Ijp7InNvdXJjZS10YWJsZSI6MiwiYWdncmVnYXRpb24iOltbImNvdW50Il1dLCJicmVha291dCI6W1siZGF0ZXRpbWUtZmllbGQiLFsiZmllbGQtaWQiLDEyXSwieWVhciJdLFsiZmstPiIsWyJmaWVsZC1pZCIsMTFdLFsiZmllbGQtaWQiLDI2XV0sWyJmay0-IixbImZpZWxkLWlkIiwxM10sWyJmaWVsZC1pZCIsNF1dXX0sImRhdGFiYXNlIjoxfSwiZGlzcGxheSI6InBpdm90IiwiZGlzcGxheUlzTG9ja2VkIjp0cnVlLCJ2aXN1YWxpemF0aW9uX3NldHRpbmdzIjp7InBpdm90X3RhYmxlLmNvbHVtbl9zcGxpdCI6eyJyb3dzIjpbWyJmay0-IixbImZpZWxkLWlkIiwxM10sWyJmaWVsZC1pZCIsNF1dLFsiZmstPiIsWyJmaWVsZC1pZCIsMTFdLFsiZmllbGQtaWQiLDI2XV1dLCJjb2x1bW5zIjpbWyJkYXRldGltZS1maWVsZCIsWyJmaWVsZC1pZCIsMTJdLCJ5ZWFyIl1dLCJ2YWx1ZXMiOltbImFnZ3JlZ2F0aW9uIiwwXV19LCJwaXZvdF90YWJsZS5jb2xsYXBzZWRfcm93cyI6WyJbXCJXaWRnZXRcIl0iLCJbXCJEb29oaWNrZXlcIl0iXX19) should open a pivot table with some rows collapsed.

Work for follow on PRs
- Collapse/expand all from the column
- Collapsing columns besides the left-most
This requires adding more subtotals first.
- Better handling for the viz setting
We should clear collapsed settings when you update the columns or change the query.

### CODE REVIEW

File: frontend/test/metabase/scenarios/visualizations/pivot_tables.cy.spec.js [hunk 1]
```diff
@@ -87,13 +87,21 @@ describe("scenarios > visualizations > pivot tables", () => {
...
+          aggregation: [["count"]],
+          breakout: [
+            ["binning-strategy", ["field-id", ORDERS.SUBTOTAL], "default"],
+          ],
+        },
+        database: 1,
+      },
+      display: "pivot",
+      visualization_settings: {},
+    });
```
Comment (paulrosenzweig) [inline frontend/test/metabase/scenarios/visualizations/pivot_tables.cy.spec.js:104]: This is unrelated, but I was updating the adjacent test anyways.

File: frontend/src/metabase/visualizations/visualizations/PivotTable.jsx [hunk 2]
```diff
@@ -393,6 +427,35 @@ export default class PivotTable extends Component {
   }
 }
 
+function RowToggleIcon({
```
Comment (paulrosenzweig) [inline frontend/src/metabase/visualizations/visualizations/PivotTable.jsx:430]: @mazameli if you want to do a styling pass, this component is the +/- icons.

**Comment by mazameli** [comment]
Awesome! FWIW, I don't think we need animation; in playing with it for a minute or two, I think animation would make things feel slower.

I think @kdoh or I will need to do some separate follow-ups for hover states and other misc. small polish, but otherwise this feels quite tight already. 🎉

File: frontend/src/metabase/visualizations/visualizations/PivotTable.jsx (hunk 2 above)
Comment (mazameli) [reply frontend/src/metabase/visualizations/visualizations/PivotTable.jsx:430]: Perfect, thanks.

Comment (mazameli) [reply frontend/src/metabase/visualizations/visualizations/PivotTable.jsx:430]: Added some styling to it. 👌🏼

**Review by mazameli — APPROVED** [review]
Seems pretty great to me on the UX side! Only thing I did notice is that on occasion, collapsing or expanding a row seemed to cause the query to get re-run, which would be great to avoid if that's possible. Otherwise 👍🏼

**Comment by mazameli** [comment]
@paulrosenzweig, do we need a code review from @kdoh or @ariya on this?

**Comment by paulrosenzweig** [comment]
> do we need a code review from @kdoh or @ariya on this?

Ideally, but if they can't get to it, we should probably merge anyways to get it into the RC.

**Review by ariya — APPROVED** [review]
LGTM!

**Pull request merged**